java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -247,4 +252,104 @@ public void set_missing_value_and_returnOldValue_is_null(BaseClient client) {
+        assertArrayEquals(new String[] {"value2", "value3"}, client.lpopCount(key, 2).get());
+        assertArrayEquals(new String[] {}, client.lrange("non_existing_key", 0, -1).get());
+        assertNull(client.lpop("non_existing_key").get());
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void lpush_lpop_lrange_type_error(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+
+        assertEquals(OK, client.set(key, "foo").get());
+
+        Exception lpushException =
+                assertThrows(ExecutionException.class, () -> client.lpush(key, new String[] {"foo"}).get());
+        assertTrue(lpushException.getCause() instanceof RequestException);
+
+        Exception lpopException = assertThrows(ExecutionException.class, () -> client.lpop(key).get());
+        assertTrue(lpopException.getCause() instanceof RequestException);
+
+        Exception lrangeException =
+                assertThrows(ExecutionException.class, () -> client.lrange(key, 0, -1).get());
+        assertTrue(lrangeException.getCause() instanceof RequestException);
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void llen_existing_non_existing_key_and_type_error(BaseClient client) {
+        String key1 = UUID.randomUUID().toString();
+        String key2 = UUID.randomUUID().toString();
+        String[] valueArray = new String[] {"value4", "value3", "value2", "value1"};
+
+        assertEquals(4, client.lpush(key1, valueArray).get());
+        assertEquals(4, client.llen(key1).get());
+        assertEquals(0, client.llen("non_existing_key").get());
+
+        assertEquals(OK, client.set(key2, "foo").get());
+
+        Exception lrangeException =
+                assertThrows(ExecutionException.class, () -> client.llen(key2).get());
+        assertTrue(lrangeException.getCause() instanceof RequestException);
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void ltrim_existing_non_existing_key_and_type_error(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        String[] valueArray = new String[] {"value4", "value3", "value2", "value1"};
+
+        assertEquals(4, client.lpush(key, valueArray).get());
+        assertEquals(OK, client.ltrim(key, 0, 1).get());
+        assertArrayEquals(new String[] {"value1", "value2"}, client.lrange(key, 0, -1).get());
+
+        // `start` is greater than `end` so the key will be removed.
+        assertEquals(OK, client.ltrim(key, 4, 2).get());
+        assertArrayEquals(new String[] {}, client.lrange(key, 0, -1).get());
+
+        assertEquals(OK, client.set(key, "foo").get());
+
+        Exception ltrimException =
+                assertThrows(ExecutionException.class, () -> client.ltrim(key, 0, 1).get());
+        assertTrue(ltrimException.getCause() instanceof RequestException);
```

Comment:
Do you want to check the error message?

Reply:
https://github.com/aws/glide-for-redis/pull/974#discussion_r1491751205 deleted them based on this comment.